### PR TITLE
Add json-serializer error handling

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR targets #635.

It only contains logging calls by now, as I'm not sure what to do when
the serialization fails. It should not fail, of course. But if that happens
we could almost do an shutdown, as the serialization should only fail if
no memory is left, and ... yeah... you get it, I guess.

What to do in case of error? Clean up, of course, but what else? Is
logging enough?

### CODE REVIEW

**Comment by neithernut** [comment]
We have to return an error of some sort, so the connection can be shut down by the connection processor.

---

The documentation of `serialize_f` btw sais: "If an internal error occured, a negative error may be returned.". The return value isn't used _yet_, but tht's another TODO (which also exists as a comment/tag in the code)

File: src/serialize/json/serializer.c [hunk 1]
```diff
@@ -492,26 +490,39 @@ serialize_value(
...
+                {
+                    int r = ws_value_set_select((struct ws_value_set*) val,
+                                                 NULL, NULL,
+                                                (int (*)(void*, void const*))
+                                                serialize_object_to_id_string,
+                                                ctx);
+
+                    if (unlikely(r < 0)) {
+                        ws_log(&log_ctx, LOG_DEBUG, "Error serializing set");
+                        //!< @todo abort serialization here?
```
Comment (neithernut) [inline src/serialize/json/serializer.c]: Yep, you should abort serialization here.

**Comment by matthiasbeyer** [comment]
The serializer already returns negative values on failure. Maybe they do not tell you anything (`-1`), but sometimes they are like `-EINVAL` or something. We should maybe change the (abstract) interface here, so that some sort of error can be returned (non-errno.h numbers).

**Comment by neithernut** [comment]
I saw that you returned `-1`, I just wanted to emphasize that all that's to be done is
- logging
- returning an error code

Returning an extensive error description is not necessary, since the functions already do logging. _maybe_ we want to return `-EIO` instead of `-1` and _maybe_ we even want to return `-EAGAIN` on some recoverable errors, but that's all I would change in the code base at this point.

**Comment by matthiasbeyer** [comment]
`maybe`? :smile: 

I'm not quite sure that there are _any_ recoverable errors here... I get a `yajl_state_error` if the parser fails and I'm then able to get an error-string describing the error. This does not help very much in manner of "recoverable" errors...

**Comment by neithernut** [comment]
In this case, return `-EIO` instead of `-1`.

**Comment by matthiasbeyer** [comment]
I checked: It makes no sense to return `-1` in any place, all occurences of `-1` in `serialize.c` were replaced by `-EIO` by the latest commit.

Feel free to suggest more changes, etc...

**Comment by neithernut** [comment]
ACK for squashing or merge, if you feel that the history is clean enough.

**Comment by matthiasbeyer** [comment]
Squashed one commit, feel free to merge.

**Pull request merged**